### PR TITLE
Micronaut performance improvements

### PR DESCRIPTION
**Pull request by dstepanov**
- More database pool threads
- Shared Netty event loop and Vertx PgClient event loop
- Let context switching
- Latest version of Gradle / JVM

### CODE REVIEW

File: frameworks/Java/micronaut/micronaut-vertx-pg-client/src/main/java/benchmark/ConnectionHolder.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+    @Property(name = "datasources.default.url") String url;
+    @Property(name = "datasources.default.username") String user;
+    @Property(name = "datasources.default.password") String password;
+    @Property(name = "datasources.default.maximum-pool-size") int maxPoolSize;
+
+    public Future<PgConnection> get() {
+        Future<PgConnection> c = conn.get();
+        if (c == null) {
+            NettyHttpRequest<?> req = (NettyHttpRequest<?>) ServerRequestContext.currentRequest().get();
```
Comment (yawkat) [inline frameworks/Java/micronaut/micronaut-vertx-pg-client/src/main/java/benchmark/ConnectionHolder.java:43]: i think i prefer ThreadExecutorMap, unless you have a reason to use the ServerRequestContext

**Review by yawkat — APPROVED** [review]

**Review by graemerocher — APPROVED** [review]

File: frameworks/Java/micronaut/micronaut-vertx-pg-client/src/main/java/benchmark/ConnectionHolder.java (hunk 1 above)
Comment (yawkat) [reply frameworks/Java/micronaut/micronaut-vertx-pg-client/src/main/java/benchmark/ConnectionHolder.java:43]: but the request is only used to get the event loop.

Comment (yawkat) [reply frameworks/Java/micronaut/micronaut-vertx-pg-client/src/main/java/benchmark/ConnectionHolder.java:43]: just cast it. i think it should work

Comment (dstepanov) [reply frameworks/Java/micronaut/micronaut-vertx-pg-client/src/main/java/benchmark/ConnectionHolder.java]: Ok

Comment (yawkat) [reply frameworks/Java/micronaut/micronaut-vertx-pg-client/src/main/java/benchmark/ConnectionHolder.java]: thanks. does it work? :sweat_smile:

**Comment by dstepanov** [comment]
@NateBrady23 Please review :)

**Pull request merged**